code/modules/mob/living/carbon/carbon_defense.dm

```diff
@@ -247,6 +247,9 @@
 		return
 
 	if(health >= 0 && !(status_flags & FAKEDEATH))
+		if(buckled && lying)
+			to_chat(M, "<span class='warning'>You need to unbuckle [src] first to do that!")
+			return
 
 		if(lying)

```

Comment:
just do if buckled under this branch

Reply:
That'll stop you from being able to hug/shake people sitting in chairs, which I think folks would complain about.

Reply:
but you're already checking it. Replace line #254 with
```
if(lying)
    if(buckled)
        to_chat(M, ...)
        return